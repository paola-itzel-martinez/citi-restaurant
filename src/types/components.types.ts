export enum ButtonVariant {
  CARD = 'CARD',
  MENU = 'MENU',
}

export enum DividerVariant {
  HORIZONTAL = 'HORIZONTAL',
  VERTICAL = 'VERTICAL',
}

export enum Portals {
  MODAL = 'MODAL',
}

export enum TypographyVariant {
  BUTTON = 'BUTTON',
  BUTTON_MENU = 'BUTTON_MENU',
  NORMAL = 'NORMAL',
  SMALL = 'SMALL',
  TITLE = 'TITLE',
}

export enum TypographyFamily {
  GENERAL = "'Roboto',sans-serif",
}

export type CSSRule = number | string;

export type CSSMarginRules = {
  m?: CSSRule;
  mx?: CSSRule;
  my?: CSSRule;
  mt?: CSSRule;
  mr?: CSSRule;
  mb?: CSSRule;
  ml?: CSSRule;
};

export type CSSPaddingRules = {
  p?: CSSRule;
  px?: CSSRule;
  py?: CSSRule;
  pt?: CSSRule;
  pr?: CSSRule;
  pb?: CSSRule;
  pl?: CSSRule;
};

export type CSSSpacingRules = {
  width?: CSSRule;
  minWidth?: CSSRule;
  maxWidth?: CSSRule;
  height?: CSSRule;
  minHeight?: CSSRule;
  maxHeight?: CSSRule;
};
