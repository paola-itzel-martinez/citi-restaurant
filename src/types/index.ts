export * from './components.types';
export * from './mediaQueries.types';
export * from './yelpAPI.types';
