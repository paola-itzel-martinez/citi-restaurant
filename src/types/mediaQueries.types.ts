export enum SCREEN_SIZE {
  DESKTOP = 'DESKTOP',
  TABLET = 'TABLET',
  MOBILE = 'MOBILE',
}
