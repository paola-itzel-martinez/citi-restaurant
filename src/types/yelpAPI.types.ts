export type Category = {
  title: string;
  alias: string;
};

export type Business = {
  id: string;
  categories: Category[];
  image_url: string;
  name: string;
  rating: number;
  url: string;
  price?: string;
};
