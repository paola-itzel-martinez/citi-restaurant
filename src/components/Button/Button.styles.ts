import styled from "styled-components";
import theme from "styled-theming";
import { MEDIA_MOBILE, palette, shadow } from "../../styled";
import { ButtonVariant, TypographyFamily } from "../../types";
import { ButtonCSS } from "./Button.types";

const { LIGHT_GRAY, NORMAL_TEXT, PRIMARY, PRIMARY_LIGHT, WHITE } = palette;

const background = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: {
    default: `linear-gradient(115deg, ${PRIMARY}, ${PRIMARY_LIGHT});`,
  },
  [ButtonVariant.MENU]: { default: LIGHT_GRAY },
});

const backgroundHover = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: PRIMARY },
  [ButtonVariant.MENU]: { default: WHITE },
});

const border = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "unset" },
  [ButtonVariant.MENU]: { default: `1px solid ${LIGHT_GRAY}` },
});

const borderHover = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "unset" },
  [ButtonVariant.MENU]: { default: `1px solid ${PRIMARY}` },
});

const color = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: WHITE },
  [ButtonVariant.MENU]: { default: NORMAL_TEXT },
});

const fontSize = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "16px" },
  [ButtonVariant.MENU]: { default: "22px" },
});

const textTransform = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "uppercase" },
  [ButtonVariant.MENU]: { default: "capitalize" },
});

const padding = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "8px 32px" },
  [ButtonVariant.MENU]: { default: "12px 32px" },
});

const wrapperShadow = theme.variants("mode", "variant", {
  [ButtonVariant.CARD]: { default: "unset" },
  [ButtonVariant.MENU]: { default: shadow.NORMAL },
});

export const Wrapper = styled.button<ButtonCSS>`
  background: ${background};
  border: ${border};
  border-radius: 8px;
  color: ${color};
  font-family: ${TypographyFamily.GENERAL};
  font-size: ${fontSize};
  font-weight: 400;
  text-transform: ${textTransform};
  box-shadow: ${wrapperShadow};
  padding: ${padding};
  white-space: nowrap;

  &.selected,
  &:hover {
    background: ${backgroundHover};
    border: ${borderHover};
    cursor: pointer;
  }

  @media ${MEDIA_MOBILE} {
    white-space: unset;
  }
`;
