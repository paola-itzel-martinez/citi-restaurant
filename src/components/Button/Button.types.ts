import { HTMLAttributes } from "react";
import { ButtonVariant } from "../../types";

export interface ButtonCSS extends HTMLAttributes<HTMLButtonElement> {
  variant: ButtonVariant;
}

export interface ButtonProps extends ButtonCSS {
  children: any;
  className?: string;
}
