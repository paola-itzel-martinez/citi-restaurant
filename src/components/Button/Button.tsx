import React from "react";
import { Wrapper } from "./Button.styles";
import { ButtonProps } from "./Button.types";

const Button = ({ children, ...otherProps }: ButtonProps) => {
  return <Wrapper {...otherProps}>{children}</Wrapper>;
};

export default React.memo(Button);
