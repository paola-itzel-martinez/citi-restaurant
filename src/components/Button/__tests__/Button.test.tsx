import React from "react";
import { customRender, fireEvent, screen } from "../../../../tests";
import { ButtonVariant } from "../../../types";
import { ButtonProps } from "../Button.types";
import { Button } from "..";

const CHILDREN = "Children";

const GeneralComponent = (props: Omit<ButtonProps, "children">) => (
  <Button {...props}>{CHILDREN}</Button>
);

describe("<Button />", () => {
  it(`Should show the label`, () => {
    customRender(<GeneralComponent variant={ButtonVariant.CARD} />);

    expect(screen.getByText(CHILDREN)).toBeInTheDocument();
  });

  it(`Should have a default css rules when variant prop is ${ButtonVariant.CARD}`, () => {
    const { container } = customRender(
      <GeneralComponent variant={ButtonVariant.CARD} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should have a default css rules when variant prop is ${ButtonVariant.MENU}`, () => {
    const { container } = customRender(
      <GeneralComponent variant={ButtonVariant.MENU} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`Should call the onClick function when its clicked`, () => {
    const MOCK_ON_CLICK = jest.fn();

    customRender(
      <GeneralComponent variant={ButtonVariant.MENU} onClick={MOCK_ON_CLICK} />
    );

    fireEvent.click(screen.getByText(CHILDREN));

    expect(MOCK_ON_CLICK).toBeCalledTimes(1);
  });
});
