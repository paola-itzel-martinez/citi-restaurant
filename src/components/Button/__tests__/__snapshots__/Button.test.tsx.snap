// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should have a default css rules when variant prop is CARD 1`] = `
.c0 {
  background: linear-gradient(115deg,#ff6f04,#ffc391);
  border: unset;
  border-radius: 8px;
  color: #ffffff;
  font-family: 'Roboto',sans-serif;
  font-size: 16px;
  font-weight: 400;
  text-transform: uppercase;
  box-shadow: unset;
  padding: 8px 32px;
  white-space: nowrap;
}

.c0.selected,
.c0:hover {
  background: #ff6f04;
  border: unset;
  cursor: pointer;
}

@media (min-width:0px) and (max-width:599px) {
  .c0 {
    text-overflow: ellipsis;
    overflow: hidden;
    padding: 8px 32px;
    white-space: unset;
  }
}

<div>
  <button
    class="c0"
  >
    Children
  </button>
</div>
`;

exports[`<Button /> Should have a default css rules when variant prop is MENU 1`] = `
.c0 {
  background: #eaeaea;
  border: 1px solid #eaeaea;
  border-radius: 8px;
  color: #415058;
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  font-weight: 400;
  text-transform: capitalize;
  box-shadow: 0px 0.25px 1px rgba(0,0,0,0.039),0px 0.85px 3px rgba(0,0,0,0.19);
  padding: 12px 32px;
  white-space: nowrap;
}

.c0.selected,
.c0:hover {
  background: #ffffff;
  border: 1px solid #ff6f04;
  cursor: pointer;
}

@media (min-width:0px) and (max-width:599px) {
  .c0 {
    text-overflow: ellipsis;
    overflow: hidden;
    padding: 8px;
    white-space: unset;
  }
}

<div>
  <button
    class="c0"
  >
    Children
  </button>
</div>
`;
